src/constraint_system/ecc/mod.rs

```diff
@@ -12,13 +12,38 @@ pub struct Point {
     x: Variable,
     y: Variable,
 }
+
+impl Point {
+    /// Return the X coordinate of the point
+    pub fn x(&self) -> &Variable {
+        &self.x
+    }
+
+    /// Return the Y coordinate of the point
+    pub fn y(&self) -> &Variable {
+        &self.y
+    }
+}
+
 /// The result of a scalar multiplication
 #[derive(Debug, Clone, Copy)]
 pub struct PointScalar {
     point: Point,
     scalar: Variable,
 }
 
+impl PointScalar {
+    /// Return the generated point
+    pub fn point(&self) -> &Point {
+        &self.point
+    }
+
+    /// Return the internal scalar
+    pub fn scalar(&self) -> &Variable {
+        &self.scalar
+    }
+}
+
```

Comment:
Not sure this is useful. The only way to create a `PointScalar` is basically doing it by yourself, which means that you already know this info. @kevaundray ?

Reply:
I assume the `PointScalar` structure is used to be passed to functions as well, therefore you might not known the original data that you used to create the structure unless you pass them as well – and it would be a waste of course.

Reply:
The scalar multiplication that is performed inside of the circuit returns this `PointScalar` type

If you already know the variable doesn’t mean you can use it as different inputs, because the internal variables that were the ones used to create the PointScalar could be different unless you somehow constrain them to the values you know.

If you have the internal variables accessible to the user, then you don’t need to perform this additional constrain